File: springboot-backend/src/main/java/com/capstone/project/service/SubmissionService.java
package com.capstone.project.service;

import com.capstone.project.exception.ResourceNotFroundException;
import com.capstone.project.model.StudySet;
import com.capstone.project.model.Submission;

import java.text.ParseException;
import java.util.List;
import java.util.Map;

public interface SubmissionService {

    List<Submission> getAllSubmissionByAssignmentId(int id);
    List<Submission> getAllSubmission();

//    Submission createSubmission(Submission submission, List<String> file_names, int type, List<String> urls, List<String> file_types);

    Submission createSubmission(Submission submission);

    Submission getSubmissionById (int id) throws ResourceNotFroundException;

//    Submission updateSubmission(int id, Submission submission, List<String> file_names, int type, List<String> urls, List<String> file_types) throws ResourceNotFroundException;

    Submission updateSubmission(int id, Submission submission) throws ResourceNotFroundException;

    Boolean deleteSubmission(int id) throws ResourceNotFroundException;

    Map<String, Object> getFilterSubmission(String search, int authorId,int assignmentId,double mark, String from, String to, String direction, int page, int size) throws ResourceNotFroundException;
}

File: react-frontend/src/services/SubmissionService.js
import axios from 'axios'

const API_BASE_URL = process.env.REACT_APP_HOST

const createSubmission = (submission) => {
    return axios.post(API_BASE_URL + '/submissions', submission)
}

const updateSubmission = (id, submission) => {
    return axios.put(API_BASE_URL + '/submissions/' + id, submission)
}

const getSubmissionById = (id) => {
    return axios.get(API_BASE_URL + '/submissions/' + id)
}

const getSubmissionByAuthorIdandAssignmentId = (authorId, assignmentId) => {
    return axios.get(
        API_BASE_URL +
            '/submissionbyauthorandassignment?authorid=' +
            authorId +
            '&assignmentid=' +
            assignmentId
    )
}

const getFilterList = (search, authorid, assignmentid, mark, page, size) => {
    return axios.get(
        API_BASE_URL +
            '/filtersubmission?search' +
            search +
            '&authorid' +
            authorid +
            '&assignmentid' +
            assignmentid +
            '&mark' +
            mark +
            '&page' +
            page +
            '&size' +
            size
    )
}

const SubmissionService = {
    getSubmissionById,
    getFilterList,
    getSubmissionByAuthorIdandAssignmentId,
    createSubmission,
    updateSubmission,
}

export default SubmissionService
